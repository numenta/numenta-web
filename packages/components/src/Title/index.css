/**
 * Numenta Web Platform and Sites source code
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2017 Numenta <http://numenta.com>
 */

._base {
  composes: black from 'tachyons';
  composes: fw6 from 'tachyons';
  composes: lh-title from 'tachyons';
  composes: mv1 from 'tachyons';
  composes: pa0 from 'tachyons';
}


.headline {
  composes: f2 from 'tachyons';
  composes: _base;
}

.title {
  composes: f3 from 'tachyons';
  composes: _base;
}
