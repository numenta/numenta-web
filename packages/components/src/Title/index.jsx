// Numenta Web Platform and Sites source code
// MIT License (see LICENSE.txt)
// Copyright © 2005—2017 Numenta <http://numenta.com>

import React from 'react'

import styles from './index.css'


const Title = ({children, headline}) => {
  const isHeadline = headline === true
  const className = isHeadline ? styles.headline : styles.title

  return (
    <h1 className={className}>
      {children}
    </h1>
  )
}

Title.propTypes = {
  children: React.PropTypes.node.isRequired,
  headline: React.PropTypes.bool.isRequired,
}

Title.defaultProps = {
  headline: false,
}

export default Title
