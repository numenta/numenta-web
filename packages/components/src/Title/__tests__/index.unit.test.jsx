import React from 'react'
import renderer from 'react-test-renderer'

import Title from '../../Title'


describe('Title React component', () => {

  it('Renders Normal correctly', () => {
    const component = renderer.create(
      <Title>
        Important Text
      </Title>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Renders Headline correctly', () => {
    const component = renderer.create(
      <Title headline={true}>
        Important Text
      </Title>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
