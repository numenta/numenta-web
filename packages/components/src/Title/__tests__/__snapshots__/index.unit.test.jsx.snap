exports[`Title React component Renders Headline correctly 1`] = `
<h1
  className="headline">
  Important Text
</h1>
`;

exports[`Title React component Renders Normal correctly 1`] = `
<h1
  className="title">
  Important Text
</h1>
`;
